helper/testhelpers/docker/testhelpers.go

```diff
@@ -23,3 +27,75 @@ func CleanupResource(t testing.T, pool *dockertest.Pool, resource *dockertest.Re
 	}
 	t.Fatalf("Failed to cleanup local container: %s", err)
 }
+
+func PrepareTestContainer(t *testing.T) (cleanup func(), retAddress, token, mountPath, keyName string, tlsConfig *api.TLSConfig) {
+	testToken, err := uuid.GenerateUUID()
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	testMountPath, err := uuid.GenerateUUID()
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+	testKeyName, err := uuid.GenerateUUID()
+	if err != nil {
+		t.Fatalf("err: %s", err)
+	}
+
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		t.Fatalf("Failed to connect to docker: %s", err)
+	}
+
+	dockerOptions := &dockertest.RunOptions{
+		Repository: "vault",
+		Tag:        "latest",
+		Cmd: []string{"server", "-log-level=trace", "-dev", fmt.Sprintf("-dev-root-token-id=%s", testToken),
+			"-dev-listen-address=0.0.0.0:8200"},
+	}
+	resource, err := pool.RunWithOptions(dockerOptions)
+	if err != nil {
+		t.Fatalf("Could not start local Vault docker container: %s", err)
+	}
+
+	cleanup = func() {
+		CleanupResource(*t, pool, resource)
+	}
+
+	retAddress = fmt.Sprintf("http://127.0.0.1:%s", resource.GetPort("8200/tcp"))
+	tlsConfig = &api.TLSConfig{
+		Insecure: true,
+	}
+
+	// exponential backoff-retry
+	if err = pool.Retry(func() error {
+		vaultConfig := api.DefaultConfig()
+		vaultConfig.Address = retAddress
+		if err := vaultConfig.ConfigureTLS(tlsConfig); err != nil {
+			return err
+		}
+		vault, err := api.NewClient(vaultConfig)
+		if err != nil {
+			return err
+		}
+		vault.SetToken(testToken)
+
+		// Set up transit
+		if err := vault.Sys().Mount(testMountPath, &api.MountInput{
```

Comment:
I feel this doesn't belong in a generic helper.  Why not move mounting of transit and creating of keys within to the test that cares about them?

Reply:
Agreed, I moved setting up Transit to its own function. Both of the functions here are still pretty transit specific, my thinking is that we can generecize as tests begin to leverage this package. The other is to make it super generic from the get-go.